notification/src/main/java/com/kyriba/conference/notification/services/DeliveryService.java

```diff
@@ -0,0 +1,115 @@
+package com.kyriba.conference.notification.services;
+
+import com.kyriba.conference.notification.dao.EmailNotificationRepository;
+import com.kyriba.conference.notification.dao.NotificationInfoRepository;
+import com.kyriba.conference.notification.dao.SmsNotificationRepository;
+import com.kyriba.conference.notification.dao.TelegaNotificationRepository;
+import com.kyriba.conference.notification.domain.EmailNotification;
+import com.kyriba.conference.notification.domain.NotificationInfo;
+import com.kyriba.conference.notification.domain.SmsNotification;
+import com.kyriba.conference.notification.domain.TelegaNotification;
+import com.kyriba.conference.notification.domain.dto.MessageStatus;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.Optional;
+import java.util.Random;
+import java.util.function.Predicate;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class DeliveryService {
+
+    private final NotificationInfoRepository notificationInfoRepository;
+
+    private final EmailNotificationRepository emailNotificationRepository;
+    private final TelegaNotificationRepository telegaNotificationRepository;
+    private final SmsNotificationRepository smsNotificationRepository;
+
+    @Transactional
+    public boolean deliveryEmailNotification(Long id) {
+        return deliveryNotification(id, this::deliveryEmail);
+    }
+
+    @Transactional
+    public boolean deliveryTelegaNotification(Long id) {
+        return deliveryNotification(id, this::deliveryTelega);
+    }
+
+    @Transactional
+    public boolean deliverySmsNotification(Long id) {
+        return deliveryNotification(id, this::deliverySms);
+    }
+
+    private boolean deliveryEmail(NotificationInfo notificationInfo) {
```

Comment:
Implementations for all 3 types of messages are completely the same. It seems you need some abstraction to apply code differently where message is processed or created.